src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj

```diff
@@ -93,6 +106,7 @@
     <Reference Include="System.Diagnostics.Debug" />
     <Reference Include="System.Diagnostics.Tracing" />
     <Reference Include="System.Memory" />
+    <Reference Include="System.Linq" />
```

Comment:
I removed the dependency of System.Linq. I was needing it for using IEnumerable<T>.ToArray. I have implemented this functionality manually. I had to enumerate IEnumerable twice to get the count and then allocate the array and copy the contents. I think this is acceptable as the perf wouldn't be a concern when performing such operation. 

By the way, there is some implementation for IEnumerable<T>.ToArray in some source file in the common folder. I didn't use this because it will need to bring a lot of code which is not worth to do. Alos, the code written there to run on net core and not for the full framework.

Reply:
If you are not particularly concerned about perfomance, the simplest way to convert IEnumerable to array is `new List(enumerable).ToArray()`.

Reply:
I thought about that but didn't do it just to avoid allocating a new List and doing the copying twice.